apps/src/code-studio/initApp/loadApp.js

```diff
@@ -66,10 +66,15 @@ export function setupApp(appOptions) {
         appOptions.app === 'weblab'
       ) {
         $('#clear-puzzle-header').hide();
-        // Only show Version History button if the user owns this project
-        if (project.isEditable()) {
-          $('#versions-header').show();
-        }
+      }
+      // If user is project owner, or teacher viewing student work
+      const isTeacher =
+        getStore().getState().currentUser?.userType === 'teacher';
+      const isViewingStudent = !!new URLSearchParams(
```

Comment:
Should be able to use util https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/code-studio/utils.js#L26.

Example: `const isViewingStudent = !!queryParams('user_id')`